packages/connection-ui/webpack.config.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+const getBaseWebpackConfig = require( '@automattic/calypso-build/webpack.config.js' );
+const DependencyExtractionWebpackPlugin = require( '@wordpress/dependency-extraction-webpack-plugin' );
+const path = require( 'path' );
+
+const isDevelopment = process.env.NODE_ENV !== 'production';
+
+const baseConfig = getBaseWebpackConfig(
+	{ WP: false },
+	{
+		entry: {}, // We'll override later
+		'output-filename': '[name].js',
+		'output-path': path.join( __dirname, './build' ),
```

Comment:
I guess this is as good as a place to discuss as any, but what do you think about how easy/hard it would be for us to use that package? I mean, the consuming package (or if we decide to combine it with the connection package, the package itself) will need to use it. We're already doing some enqueuing, asset loading, etc. Should we be putting the built files inside the package itself, or should we be using the built Jetpack folder?
Basically what I'm asking is **What would a release of this package look like**?

Reply:
There are going to be two files in the `./build` directory: `index.js` and `index.css`.
The plan is to load them both when Connection Manager UI page is requested: https://github.com/Automattic/jetpack/pull/18303/files#diff-e56f98a9a60baeb985dc0456cfe3c301ed65bc05c050d1c62250bbc589487d19R56-R61

> should we be using the built Jetpack folder

I'm not sure I understand the idea, could you elaborate.

> What would a release of this package look like?

The release process wouldn't change a bit. The package is getting build by the existing command `yarn build-packages`, which in turn is a part of the `yarn build`.
The package will have its own `./build` directory, and its own `enqueue` calls in the PHP code to load them on demand.
The only downside of this approach I see is that each plugin will have its own copy of the built JS and CSS files, but they aren't going to be too big to worry about the size.

Plugins other than Jetpack will need to run `yarn --cwd ./packages/connection-ui build` to build the package, or somehow integrate it into their own build system.

I'm also wrestling with an idea to remove the `./_inc` directory with the actual JS source code while building a package for a release. That would make the package slimmer.
A command like `yarn --cwd ./packages/connection-ui build-release` would run the regular build, and run the `rm -rf _inc` afterwards.

Please let me know what you think, as I might be missing some underlying issues.

Reply:
To add on to https://github.com/Automattic/jetpack/pull/18303#issuecomment-758347165 , where GC is going is that after a PR is merged to `master` that changed a package file, in addition to committing the change to the mirror repo, it will also run a build step (see https://github.com/Automattic/jetpack/pull/18303#issuecomment-758347165 for where it's at now, but it'll change as we have more than Jetpack the plugin using it).

The idea here in Composer, a consuming project shouldn't need to do any building of JS.

I'm open to other ideas and this is something that we added in as part of the reorg while I was working on getting a built version of Jetpack copied over to the new jetpack-production repo.

Reply:
We've had a somehow similar discussion in #17571 and #17562, but went in a different direction. It may be good to agree on a common solution for all our packages.

> Plugins other than Jetpack will need to run yarn --cwd ./packages/connection-ui build to build the package, or somehow integrate it into their own build system.

I think this may be problematic for some. I think part of the assumptions behind PHP packages is that just adding them as a dependency for your project gives you access to everything they have to offer, without having to run commands afterwards to edit the contents of the `vendor/` directory in your app.

Reply:
Hi Jeremy, Kraft.

These are very good points. Let's wait for the aforementioned repo reorganizing to be merged then, and go from there.

Reply:
So the repo reorg has been merged, and I rebased the PR to include them.

> our repo reorg patch that, will among other things, allow for individual projects--like this new package--to build and commit to their mirror repo a JS-built version.

That seems to be the ideal scenario, publishing a built version into the package repo, so it could be included into other plugins without any need for building the JS.
@kraftbj, is there any documentation on publishing pre-built packages I could review?

Reply:
Sorry for missing this comment until now! Is there a mirror repo already created for this one? If so, we'd need to add it to the `composer.json`. See lazy-images for an example: https://github.com/Automattic/jetpack/blob/master/projects/packages/lazy-images/composer.json#L45

With a `build-production` and a mirror-repo set, the GH action will automatically build it and put the built version in the mirror repo.